src/libraries/System.Private.CoreLib/src/System/Reflection/Nullability/NullabilityInfoContext.cs

```diff
@@ -0,0 +1,430 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Reflection
+{
+    public sealed class NullabilityInfoContext
+    {
+        private const string CompilerServicesNameSpace = "System.Runtime.CompilerServices";
+        private readonly Dictionary<Module, NotAnnotatedStatus> _publicOnlyModules = new();
+        private readonly Dictionary<MemberInfo, NullabilityState> _context = new();
+        private static readonly NullabilityInfo[] s_emptyArray = Array.Empty<NullabilityInfo>();
+
+        [Flags]
+        private enum NotAnnotatedStatus
+        {
+            None = 0x0, // no restriction, all members annotated
+            Private = 0x1, // private members not annotated
+            Internal = 0x2 // internal members not annotated
+        }
+
+        private NullabilityState GetNullableContext(MemberInfo? memberInfo)
+        {
+            while (memberInfo != null)
+            {
+                if (_context.TryGetValue(memberInfo, out var state))
+                {
+                    return state;
+                }
+
+                var attributes = memberInfo.GetCustomAttributesData();
+                foreach (var attribute in attributes)
+                {
+                    if (attribute.AttributeType.Name == "NullableContextAttribute" &&
+                        attribute.AttributeType.Namespace == CompilerServicesNameSpace &&
+                        attribute.ConstructorArguments.Count == 1)
+                    {
+                        state = TranslateByte(attribute.ConstructorArguments[0].Value);
+                        _context.Add(memberInfo, state);
+                        return state;
+                    }
+                }
+
+                memberInfo = memberInfo.DeclaringType;
+            }
+
+            return NullabilityState.Unknown;
+        }
+
+        [RequiresUnreferencedCode("Nullability attributes are trimmed by the linker")]
+        public NullabilityInfo Create(ParameterInfo parameterInfo)
+        {
+            if (parameterInfo.Member is MethodBase method &&
+                (method.IsPrivate || method.IsFamilyAndAssembly || method.IsAssembly))
+            {
+                if (IsPublicOnly(method.IsPrivate, method.IsFamilyAndAssembly, method.IsAssembly, method.Module))
```

Comment:
When the member is private or internal nullability attributes might be omitted, we need to check module-level  `NullablePublicOnlyAttribute`, and if the attribute is set just return `NullabilityState.Unknown`

Reply:
This info would be valuable in a C# comment.